### PR TITLE
CaptionModel v1 (WIP)

### PR DESCRIPTION
**Pull request by johnx25bd**
Karim's changes, plus some fixes to attention masks.

### CODE REVIEW

File: api/models/vision_transformer.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    dummy_input_ids = torch.randint(0, gpt2_vocab_size, (batch_size, seq_len))  # Random token IDs for GPT-2
+    dummy_attention_mask = torch.ones_like(dummy_input_ids, dtype=torch.bool)  # Attention mask
+
+    # Initialize the full model
+    model = CaptionModel()
+
+    # Preprocess images using the encoder's transform
+    transform = model.encoder.transforms
+    transformed_images = transform(dummy_images) # (batch_size, 3, 224, 224)
+
```
Comment (johnx25bd) [inline api/models/vision_transformer.py:141]: Add patching + flattening before the image is passed into the encoder

**Review by samrae7 — APPROVED** [review]
